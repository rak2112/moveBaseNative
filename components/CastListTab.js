//@flow
import React, { PureComponent } from 'react';
import MoviesPage from './MoviesPage';
import IosListLayout from './IosListLayout';

type Props = {
  cast: Array<{
    id: number,
    name: string
  }>,
  // renderListItems(Object): React$Node,
  // keyExtractor({id: number}, string):number,
};

const keyExtractor = (item, index): number => item.id;

class CastListTab extends PureComponent <Props> {
  renderListItems = ({item}: Object): React$Node => (
    <IosListLayout item={item}/>
  );
  render() {
    const {cast: items} = this.props;
    return (
      <MoviesPage
        renderListItems={this.renderListItems}
        keyExtractor={keyExtractor}
        items = {items}/>
    );
  }
};


export default CastListTab;
