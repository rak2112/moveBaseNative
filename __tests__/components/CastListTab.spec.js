import 'react-native';
import React from 'react';
import CastListTab from './../../components/CastListTab';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

fit('renders correctly', () => {
  const tree = renderer.create(<CastListTab />).toJSON();
  expect(tree).toMatchSnapshot();
});
