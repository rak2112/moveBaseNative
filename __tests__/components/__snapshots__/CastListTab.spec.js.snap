// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#323232",
      "flex": 1,
      "height": "auto",
    }
  }
  virtual={undefined}
>
  <RCTScrollView
    ListFooterComponent={null}
    data={undefined}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={null}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={undefined}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
